flink-connector-cassandra/src/test/java/org/apache/flink/connector/cassandra/source/CassandraTestEnvironment.java

```diff
@@ -115,51 +134,30 @@ public void tearDown() throws Exception {
         stopEnv();
     }
 
+    private static void addJavaOpts(GenericContainer<?> container, String... opts) {
+        String jvmOpts = container.getEnvMap().getOrDefault("JVM_OPTS", "");
+        container.withEnv("JVM_OPTS", jvmOpts + " " + StringUtils.join(opts, " "));
+    }
+
     private void startEnv() throws Exception {
-        // need to start the container to be able to patch the configuration file inside the
-        // container
-        // CASSANDRA_CONTAINER#start() already contains retrials
+        // configure container start to wait until cassandra is ready to receive queries
+        cassandraContainer.waitingFor(new CassandraQueryWaitStrategy());
+        // start with retrials
         cassandraContainer.start();
         cassandraContainer.followOutput(
                 new Slf4jLogConsumer(LOG),
                 OutputFrame.OutputType.END,
                 OutputFrame.OutputType.STDERR,
                 OutputFrame.OutputType.STDOUT);
-        raiseCassandraRequestsTimeouts();
-        // restart the container so that the new timeouts are taken into account
-        cassandraContainer.stop();
-        cassandraContainer.start();
+
         cluster = cassandraContainer.getCluster();
         clusterBuilder =
                 createBuilderWithConsistencyLevel(
                         ConsistencyLevel.ONE,
                         cassandraContainer.getHost(),
                         cassandraContainer.getMappedPort(CQL_PORT));
 
-        int retried = 0;
```

Comment:
You can actually drop these retries and let testcontainers do this using a wait strategy, e.g. `CassandraQueryWaitStrategy`

Reply:
There were some flaky `NoHostAvailableException` in case of load on the jenkins server hence this addition, I hope the `CassandraQueryWaitStrategy` is not sensitive to load. We'll see in time when the source PR is merged if the tests are flaky again or not